### ISSUE TITLE
startdde needs more than 25% cpu

### ISSUE THREAD

**Issue by LinuxUserGD** [linuxdeepin/startdde#8]
I'm running Antergos Linux 15.5 Desktop 64 Bit with Deepin.
While I don't have any other programs open, the deepin system monitor shows ~30% cpu usage (25% from startdde). I've tried to kill the process which ends in a black screen.
Is there any workaround to fix the high amount of cpu usage?

**Comment by kameleon83** [comment]
Same problem : 12,5% with core i7

**Comment by LinuxUserGD** [comment]
Core i5 with Intel Graphics 620 (OpenGL ES 3.0 mesa drivers). Would I get a message from the system if it uses software rendering? I changed the distro to Budgie and now the CPU is at 2% while I'm still running deepin apps like the file and the process manager. Does 25% and 12,5% mean that it runs 100% in one thread / one core? Can't check it yet.

**Comment by Pan5ky** [comment]
Same problem, 100% usage on core  i7 7500u

**Comment by OmarSquircleArt** [comment]
The same happening with my Ryzen 7 1700. Startdde is taking up 6.2% which is exactly one thread from the 16 threads I have. There is another report with the same issue [here](https://github.com/linuxdeepin/developer-center/issues/496).

**Comment by jouyouyun** [comment]
@electricface

**Comment by ekhaled667** [comment]
[Here is perf data and startdde journal.](https://github.com/linuxdeepin/startdde/files/1871393/perf_data.tar.gz)

**Comment by maxcruz** [comment]
After starting the session on Arch Linux startdde consumes the 100% of the CPU

**Comment by jouyouyun** [comment]
@ekhaled667 download the [file](https://drive.google.com/open?id=1TeigcjT63XpGu9M9xpQUvVjOEFxNm_mR), then copy to `/usr/bin/startdde` and exec `chmod +x /usr/bin/startdde`.
If the bug ocurred again, please update perf.data and dde.log, thanks!

**Comment by ekhaled667** [comment]
The problem persists.
[Here is the new log.](https://github.com/linuxdeepin/startdde/files/1879968/perfdata.tar.gz)

**Comment by epoiate** [comment]
[Here](https://github.com/linuxdeepin/startdde/files/1880018/perf.tar.gz)
Actually, now it worse. My dock won't load, reverted the file and it still won't load... how can i fix this?

**Comment by DasJott** [comment]
I also experience that sometimes. Dock sometimes takes **minutes** to appear on screen. With normal version.

**Comment by epoiate** [comment]
Yeah, I fixed it already. It's normal for it to take a few seconds (I'm using a SSD), but in my case it wasn't loading at all. A lot of things wasn't working too.

**Comment by DasJott** [comment]
Using SSD here too. Precisely it is a Dell XPS13 here. And it still sometimes takes minutes.
Also while runtime dde-session-daemon jumps up to 400% which drains battery big time. I don't know, if that is related...

**Comment by ekhaled667** [comment]
@jouyouyun Yes, Always.

**Comment by epoiate** [comment]
@snyh 
Here is my results 
[dde_strace.log](https://github.com/linuxdeepin/startdde/files/1886847/dde_strace.log)

PS: startdde is ALWAYS using a full core of my 6600K
![full core](https://user-images.githubusercontent.com/6474351/38461655-7ded56cc-3aac-11e8-989b-ea69f887ce8f.png)

**Comment by snyh** [comment]
@epoiate The strace's (syscall) log hasn't any problem. 

Please wait a moment, @jouyouyun will build and upload a startdde with golang perf enable.

**Comment by electricface** [comment]
https://golang.org/pkg/net/http/pprof/
port is 6060

https://pan.baidu.com/s/1BnR7QIPKSD8a4skZxbe_Mw

**Comment by epoiate** [comment]
@electricface can you elaborate the exact steps that I should take?

**Comment by OmarSquircleArt** [comment]
@electricface @jouyouyun Is there an update on this? The problem is still there and such CPU usage getting tedious.

**Comment by Alpkant** [comment]
This problem is happening to me as well and if i kill the startdde from htop, it restarts the environment and i log again and cpu usage returns to normal values. However lot of time it uses 100% of the one of my cores until i kill the startdde.

**Comment by jouyouyun** [comment]
Try to edit `/etc/default/grub` to add parameter `vsyscall=emulate`, then exec `grub-mkconfig -o /boot/grub/grub.cfg` to update, then reboot.

**Comment by Alpkant** [comment]
@jouyouyun I didn't see any high usage after editing grub file. Thank you.

**Comment by ekhaled667** [comment]
@jouyouyun It didn't work for me.

By the way, I'm on Arch Linux not Deepin OS.

And I don't use GRUB, I use systemd-boot.

**Comment by OmarSquircleArt** [comment]
@jouyouyun Same here, that does work. Arch Linux and don't use GRUB.

**Comment by jouyouyun** [comment]
:sob:

**Comment by ekhaled667** [comment]
For me, It's always utilizing one thread.
Even if I log out and then log in again, in 5 seconds it starts using 100% of one thread.
Even if I kill the process.
Even if I reboot.
Not once did it not utilize the thread.

**Comment by msusenburger** [comment]
Had this issue the last month at least. Changed today from launching via startx to launching via lightdm and issues seems resolved for me.

**Comment by OmarSquircleArt** [comment]
It is always happening for me as well. Always utilizing one thread. Tried rebooting, killing and restarting and nothing worked.

**Comment by jouyouyun** [comment]
Have you install `gnome-keyring`?

**Comment by jouyouyun** [comment]
Try this [startdde](https://drive.google.com/file/d/1Uc8Ksb3LEW0BvYuNI5n4Tf6vmWpNK_6M/view?usp=sharing), downloaded and `chmod +x`, then copy to `/usr/bin/startdde`.

**Comment by HugeGameArtGD** [comment]
I've updated to 3.1.29-1 on Arch Linux without the patch and it works for me :+1:
The deepin system monitor shows around 3% cpu usage.

**Comment by LinuxUserGD** [comment]
I've updated to 3.1.29-1 on Arch Linux without the patch and it works for me :+1:
The deepin system monitor shows around 3% cpu usage.

**Comment by msusenburger** [comment]
Version 3.1.29-1 is also working in my case, even after launching with `startx`.
`gnome-keyring` is installed in version 1:3.28.2-1.

Thanks for fixing :+1:

**Comment by OmarSquircleArt** [comment]
Just tried the previously mentioned startdde file. It seems to be working fine now. Thanks !

**Comment by ekhaled667** [comment]
> Try this startdde, downloaded and chmod +x, then copy to /usr/bin/startdde.

  Note: you should update your system firstly, startdde version should be 3.1.29-1 after updated.

This worked for me too, thanks for your time.

**Comment by jouyouyun** [comment]
Fixed in  `startdde 3.1.30`, please update!

**Comment by Stakdek** [comment]
Running on Ubuntu 18.04 I can't update startdde to `3.1.30`. It stops at `3.1.28-1`. I have the same high CPU issue.
